logic/access_point_formspec.lua

```diff
@@ -189,6 +189,15 @@ local function get_access_point_formspec(pos, invName, optMeta, playerName)
 end
 
 local function show_access_point_formspec(pos, playerName, optMeta)
+  if minetest.get_modpath("mcl_core") then
+    local player = minetest.get_player_by_name(playerName)
+    if not player then return end
+    local inv = player:get_inventory()
+    if inv then
+      inv:set_width("craft", 3)
+      inv:set_size("craft", 9)
+    end
```

Comment:
This is permanently going to change the MCL inv size isn't it? Any negative side-effects from that? Your main inv crafting gird in MCL only shows 2x2 right? so the items in the other 5 slots will still be there but hidden?

Reply:
No, I checked. Every time you open the normal inventory formspec, it sets the size to 2x2, and every time you open a crafting table, it sets it to 3x3. This is how MCL does it already.

Every time you exit the MCL crafting formspec, your crafting grid is emptied (either by returning items to your inventory or dropping them if there's not enough room). I suppose this is something that needs to be added.

Reply:
If it sets the size to 2x2, it means it will delete items that were left in the crafting grid - so yes, you would have to find a way to drop the items out of the Access Point. 

Do be aware that the Access Point UI is used by the Wireless Access pad, the exact same code is called as though the user right-clicked an Access Point (this is why I use detached inventories for most of it, and not node inventories). 

So if you're dropping items once the user closes an access point, they are best dropped at the user's position, rather than the access point's position.

Reply:
Just found this in MineClone2's code:
```
-- Drop items in craft grid and reset inventory on closing
minetest.register_on_player_receive_fields(function(player, formname, fields)
	if fields.quit then
		mcl_util.move_player_list(player, "craft")
		mcl_util.move_player_list(player, "enchanting_lapis")
		mcl_util.move_player_list(player, "enchanting_item")
		if not minetest.is_creative_enabled(player:get_player_name()) and (formname == "" or formname == "main") then
			set_inventory(player)
		end
	end
end)
```

So it looks like it clears the crafting grid upon closing *any* formspec, which is convenient.